src/common/components/PlaceholderFactory.tsx

```diff
@@ -10,9 +10,14 @@ interface PlaceholderInterface {
 // thrown in that case.
 export const PlaceholderFactory = (name: string) => {
   const Placeholder: FC<PlaceholderInterface> = (props) => {
+    const previewDomains = new Set([
+      '',
+      'ci-europa.kbase.us',
+      'narrative-dev.kbase.us',
+    ]);
     const invalidEnvironment =
       process.env.NODE_ENV === 'production' &&
-      process.env.REACT_APP_KBASE_ENV !== 'ci-europa';
+      !previewDomains.has(process.env.REACT_APP_KBASE_DOMAIN || '');
```

Comment:
This could be managed as configuration in the `build_deploy.sh` script, but adding a preview environment in either case would still require a code change. The other option is to manage this via an environment variable or a repository variable.